ReactAndroid/src/main/java/com/facebook/react/views/art/ARTShapeShadowNode.java

```diff
@@ -164,6 +170,52 @@ protected boolean setupStrokePaint(Paint paint, float opacity) {
     return true;
   }
 
+  /*
+   * calculate stops count from array
+   * from array minus first 5 elements because it 1 - LinearGradient key, 2,3,4,5 - LinearGradient position
+   * and after devide on 5 because 1,2,3,4 - colors A,R,G,B and 5 - stop per 4 colors.
+   */
+  private static int getStopsCount(float[] value) {
+    int count = (value.length - 5)/5;
+    return count;
+  }
+
+  /*
+   * sorting stops and stopsColors from array
+   */
+  private static int gradientStopsParsing(float[] value, int stopsCount, float[] stops, int[] stopsColors) {
+    int startStops = value.length - stopsCount;
+    int startColorsPosition = 5;
+    for (int i = 0; i < stopsCount; i++) {
+      stops[i] = i == 0 ?
+        value[startStops + i] :
```

Comment:
This doesn't make sense to me. You check if `i==0` so there is no point of adding `i` to `startStops`

Seems like you can greatly simplify this whole part and you shouldn't have that many conditionals:
For `stops[i]` you perhaps want to assign sth like `value[startStops + i*5]` and similarity for the colors